sound/soc/sof/intel/hda-dsp.c

```diff
@@ -239,3 +239,16 @@ int hda_dsp_core_reset_power_down(struct snd_sof_dev *sdev,
 	return ret;
 }
 
+int hda_dsp_suspend(struct snd_sof_dev *sdev, int state)
+{
+	/* power down DSP */
+	return hda_dsp_core_reset_power_down(sdev, HDA_DSP_CORE_MASK(0) |
+					     HDA_DSP_CORE_MASK(1));
+}
+
+int hda_dsp_resume(struct snd_sof_dev *sdev)
+{
+	/* power up the DSP */
+	return hda_dsp_core_power_up(sdev, HDA_DSP_CORE_MASK(0) |
+				     HDA_DSP_CORE_MASK(1));
```

Comment:
Btw, we need to be able to handle different core masks, e.g. APL has 2, CNL has 4.

Reply:
@lgirdwood sure. I have tested this to work only for APL as yet. I will make the changes for CNL next.